src/addagent/validate.c

```diff
@@ -30,6 +32,8 @@
     #define Privsep_GetGroup(x) -1
 #endif
 
+static const char *SQL_SELECT_KEEPALIVE = "global sql SELECT last_keepalive FROM agent WHERE name = '%s';";
```

Comment:
here you should say `WHERE name = '%s' AND ip = '%s';`

Reply:
The QUERY uses only the name of the agent, the parameter `ip` is unused

Reply:
`// Unused
    (void) ip;
`

Reply:
What about if we have two agents with the same name? The only way to differentiate them is by using also the IP address. I think that's why that information was used in the name of the `agent-info` files.
Shouldn't we also make use of the IP?

Reply:
Query updated with the same structure than **SQL_FIND_AGENT**

Reply:
Perfect!